File: ObamaClient/src/app/model/comment.ts
import {User} from './user';

export class Comment {
  get user(): User {
    return this._user;
  }

  get content(): string {
    return this._content;
  }

  get id(): number {
    return this._id;
  }

  private _user: User;
  private _content: string;
  private _id: number;

  constructor(user: User, content: string, id: number) {
    this._user = user;
    this._content = content;
    this._id = id;
  }

}

File: graubunden/src/main/java/nu/obama/graubunden/model/Comment.java
package nu.obama.graubunden.model;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;

@Entity
@Table(name = "\"comments\"")

public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "users")
    private User user;

    @ManyToOne
    @JoinColumn(name = "posts")
    private Post post;

    @NotBlank
    private String content;

    public Comment(){

    }

    public Comment (String content){
        this.content = content;
    }

    public Long getId() {
        return id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user){
        this.user = user;
    }

    public Post getPost() {
        return post;
    }

    public void setPost(Post post) {
        this.post = post;
    }


    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}
